File: Samples/State Pattern/Sample_0 - CS/RunState.cs
namespace Sample
{
    internal class RunState : State
    {
        public RunState()
        {
            _name = "RunState";
        }

        public override void Enter()
        {
            Console.WriteLine("RunState called Enter");
        }

        public override void Execute()
        {
            Console.WriteLine("RunState called Execute");
        }

        public override void Exit()
        {
            Console.WriteLine("RunState called Exit");
        }
    }
}

File: Samples/State Pattern/Sample_0 - CPP/RunState.cpp
#include "RunState.h"

void RunState::Enter()
{
	cout << "RunState called Enter!" << endl;
}

void RunState::Execute()
{
	cout << "RunState called Execute!" << endl;
}

void RunState::Exit()
{
	cout << "RunState called Exit!" << endl;
}
